worlds/shorthike/Rules.py

```diff
@@ -15,19 +16,19 @@ def create_rules(self, location_table):
         # Shop Rules
         if loc["purchase"] and not options.coins_in_shops:
             forbid_items_for_player(multiworld.get_location(loc["name"], player), self.item_name_groups['Coins'], player)
-        if loc["purchase"] >= get_min_shop_logic_cost(self) and options.shop_check_logic != 0:
-            if options.shop_check_logic == 1 or options.shop_check_logic == 2:
+        if loc["purchase"] >= get_min_shop_logic_cost(self) and options.shop_check_logic != ShopCheckLogic.option_nothing:
+            if options.shop_check_logic == ShopCheckLogic.option_fishing_rod or options.shop_check_logic == ShopCheckLogic.option_fishing_rod_and_shovel:
                 add_rule(multiworld.get_location(loc["name"], player),
                     lambda state: state.has("Progressive Fishing Rod", player))
-            if options.shop_check_logic == 3 or options.shop_check_logic == 4:
+            if options.shop_check_logic == ShopCheckLogic.option_golden_fishing_rod or options.shop_check_logic == ShopCheckLogic.option_golden_fishing_rod_and_shovel:
                 add_rule(multiworld.get_location(loc["name"], player),
                     lambda state: state.has("Progressive Fishing Rod", player, 2))
-            if options.shop_check_logic == 2 or options.shop_check_logic == 4:
+            if options.shop_check_logic == ShopCheckLogic.option_shovel or options.shop_check_logic == ShopCheckLogic.option_fishing_rod_and_shovel or options.shop_check_logic == ShopCheckLogic.option_golden_fishing_rod_and_shovel:
```

Comment:
If you have a bit more patience for me:

With this option comparison type, you can safely do:

`if options.shop_check_logic in {ShopCheckLogic.option_shovel, ShopCheckLogic.option_fishing_rod_and_shovel, ShopCheckLogic.option_golden_fishing_rod_and_shovel}:`

which would be a bit shorter :)
(This line does now violate Pep8 due to being over 120 characters long. Which is not the end of the world, but there is a shorter option)

Reply:
(Ideally this would be done everywhere where you currently have `option == A or option == B`)

Reply:
Done